libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,181 @@
+import {
+    AfterContentInit,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    ContentChild,
+    ViewChild,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Injector,
+    Input,
+    OnDestroy,
+    Optional,
+    Output,
+    ViewEncapsulation, TemplateRef, ContentChildren
+} from '@angular/core';
+import { DynamicComponentService } from '../utils/dynamic-component/dynamic-component.service';
+import { ActionSheetBodyComponent } from './action-sheet-body/action-sheet-body.component';
+import { ActionSheetControlComponent } from './action-sheet-control/action-sheet-control.component';
+import {
+    FocusEscapeDirection,
+    KeyboardSupportService
+} from '../utils/services/keyboard-support/keyboard-support.service';
+import { ActionSheetItemComponent } from './action-sheet-item/action-sheet-item.component';
+import { startWith, takeUntil } from 'rxjs/operators';
+import { merge, Subject } from 'rxjs';
+import {ActionSheetMobileComponent} from './action-sheet-mobile/action-sheet-mobile.component';
+import {ACTION_SHEET_COMPONENT} from './action-sheet.interface';
+
+@Component({
+    selector: 'fd-action-sheet',
+    templateUrl: './action-sheet.component.html',
+    styleUrls: ['./action-sheet.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        KeyboardSupportService
+    ]
+})
+export class ActionSheetComponent implements AfterContentInit, AfterViewInit, OnDestroy {
+
+    /** Whether should be displayed in compact mode **/
+    @Input()
+    compact = false;
+
+    /** Whether should be displayed in mobile mode **/
+    @Input()
+    mobile = false;
+
+    /** Whenever links should be visible **/
+    @Input()
+    open = false;
+
+    /** Whether internal keyboard support should be enabled. It's enabled by default */
+    @Input()
+    keyboardSupport = true;
+
+    /** To allow user to determine what event he wants to trigger the messages to show
+     * Accepts any [HTML DOM Events](https://www.w3schools.com/jsref/dom_obj_event.asp).
+     **/
+    @Input()
+    triggers: string[] = ['click'];
+
+
+    /** Event thrown, when focus escapes the list */
+    @Output()
+    focusEscapeList = new EventEmitter<FocusEscapeDirection>();
+
+    /** Event emitted, when the combobox's popover body is opened or closed */
+    @Output()
+    readonly openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    @ContentChild(ActionSheetBodyComponent) actionSheetBody;
+
+    // @ContentChildren(ActionSheetItemComponent, { descendants: true }) actionSheetBody;
+
+    /** @hidden */
+    @ViewChild('actionSheetBodyTemplate')
+    actionSheetBodyTemplate: TemplateRef<any>;
+
+    @ContentChild(ActionSheetControlComponent) actionSheetControl;
+    @ContentChild(ActionSheetMobileComponent) actionSheetMobile;
```

Comment:
This one can be removed. Also ActionSheetMobileComponent has reference passed as view, not content